packages/blocks/shuffle/src/components/item.tsx

```diff
@@ -1,123 +1,146 @@
-import React, { FunctionComponent, useState } from "react";
-import AddIcon from "@mui/icons-material/Add";
+import React, {
+  forwardRef,
+  useState,
+  useEffect,
+  CSSProperties,
+  RefObject,
+} from "react";
 import CloseIcon from "@mui/icons-material/Close";
 import DragIndicatorIcon from "@mui/icons-material/DragIndicator";
 import { Box, IconButton, ListItem, TextField, Paper } from "@mui/material";
-import { useSortable } from "@dnd-kit/sortable";
-import { CSS } from "@dnd-kit/utilities";
+import { DraggableAttributes } from "@dnd-kit/core";
 
-type ItemProps = {
+export type ItemProps = {
   id: string;
   value: string;
   isDragging?: boolean;
   onValueChange?: (value: string) => void;
   onItemBlur?: () => void;
-  onAdd?: () => void;
   onDelete?: () => void;
+  paperStyle?: CSSProperties;
+  attributes?: DraggableAttributes;
+  listeners?: Record<string, Function>;
+  style?: CSSProperties;
+  dragOverlay?: RefObject<HTMLDivElement>;
+  ref?: (node: HTMLElement | null) => void;
 };
 
-export const Item: FunctionComponent<ItemProps> = ({
-  id,
-  value,
-  isDragging,
-  onValueChange,
-  onItemBlur,
-  onAdd,
-  onDelete,
-}) => {
-  const { attributes, listeners, setNodeRef, transform, transition } =
-    useSortable({ id });
+export const Item = forwardRef<HTMLLIElement, ItemProps>(
+  (
+    {
+      value,
+      isDragging,
+      onValueChange,
+      onItemBlur,
+      onDelete,
+      paperStyle,
+      attributes,
+      style,
+      listeners,
+      dragOverlay,
+    },
+    ref,
+  ) => {
+    const [isHovered, setIsHovered] = useState(false);
 
-  const style = {
-    transform: CSS.Transform.toString(transform),
-    transition,
-  };
+    useEffect(() => {
+      if (dragOverlay?.current) {
+        dragOverlay.current.animate(
+          [
+            {
+              transform: "scale(1)",
+            },
+            {
+              transform: "scale(1.05)",
+            },
+          ],
+          {
+            duration: 250,
+            easing: "ease",
+            fill: "forwards",
+          },
+        );
+      }
+    }, [dragOverlay]);
```

Comment:
This being here is… strange to me. Effects can fire more often than you'd think and this seems to be a "one time thing". Maybe the animation ought to be triggered by an event in the parent node or something?

Reply:
Moved the animation to the sx prop in 66778c9f38a66288594b4146280823c154fdd0b5.